keps/sig-node/4381-dra-structured-parameters/README.md

```diff
@@ -1188,239 +937,309 @@ drivers are expected to be written for Kubernetes.
 
 ##### ResourceSlice
 
-For each node, one or more ResourceSlice objects get created. The kubelet
-publishes them with the node as the owner, so they get deleted when a node goes
+For each node, one or more ResourceSlice objects get created. The drivers
+on a node publish them with the node as the owner, so they get deleted when a node goes
 down and then gets removed.
 
 All list types are atomic because that makes tracking the owner for
 server-side-apply (SSA) simpler. Patching individual list elements is not
-needed and there is a single owner (kubelet).
+needed and there is a single owner.
 
 ```go
-// ResourceSlice provides information about available
-// resources on individual nodes.
+// One of more slices represent a pool of devices managed by a given driver. For
+// node-local devices, each node is a pool. How many slices the driver uses
+// to publish that pool is driver-specific. Within each pool the device name must
+// be unique.
+//
+// In which slice a device gets published may change over time. The unique identifier
+// for a node-local device is the tuple `<driver name>/<node name>/<device name>`. Each
+// of these names is a DNS label or domain, so it is okay to concatenate them
+// like this in a string with a slash as separator.
+//
+// Whenever a driver needs to update a pool, it bumps the pool generation number
+// and updates the slices with that new number and new device definitions. A consumer
+// must only use device definitions from the slices with the highest generation number
+// and ignore the others.
+//
+// If necessary, a consumer can check the number of total devices in a pool (included
+// in each slice) to determine whether its view of a pool is complete.
+//
+// For devices that are not local to a node, the node name is not set. Instead,
+// a pool name is chosen by the driver if necessary. It can be left blank if the
+// device names are unique in the entire cluster. The unique identifier then
+// has the same format as before with `<driver name>[/<pool name>]/<device name>`.
```

Comment:
```suggestion
// The slice in which a device gets published may change over time. The unique identifier
// for a node-local device is the tuple `<driver name>/<node name>/<device name>`. Each
// of these names is a DNS label or domain, so it is okay to concatenate them
// like this in a string with a slash as separator.
//
// Whenever a driver needs to update a pool, it bumps the pool generation number
// and updates all slices with that new number and any new device definitions. A consumer
// must only use device definitions from slices with the highest generation number
// and ignore all others.
//
// If necessary, a consumer can check the number of total devices in a pool (included
// in each slice) to determine whether its view of a pool is complete.
//
// For devices that are not local to a node, the node name is not set. Instead,
// a pool name is chosen by the driver as necessary. It can be left blank if the
// device names are unique in the entire cluster. The unique identifier then
// has the same format as before with `<driver name>[/<pool name>]/<device name>`.
```